adapters/sleep_test.go

```diff
@@ -0,0 +1,70 @@
+package adapters_test
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/adapters"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_SleepDurationUnmarshalJSON(t *testing.T) {
+	var duration adapters.SleepDuration
+	err := duration.UnmarshalJSON([]byte(`"30s"`))
+	assert.NoError(t, err)
+	assert.Equal(t, adapters.SleepDuration(30*time.Second), duration)
+
+	err = duration.UnmarshalJSON([]byte(`"1m"`))
+	assert.NoError(t, err)
+	assert.Equal(t, adapters.SleepDuration(1*time.Minute), duration)
+
+	// no quotes
+	err = duration.UnmarshalJSON([]byte("30s"))
+	assert.Error(t, err)
+
+	// no units
+	err = duration.UnmarshalJSON([]byte(`"30"`))
+	assert.Error(t, err)
+
+	// garbage
+	err = duration.UnmarshalJSON([]byte(`"xxx"`))
+	assert.Error(t, err)
+}
+
+func TestSleep_Perform(t *testing.T) {
+	tests := []struct {
+		name         string
+		params       string
+		parseErrored bool
+		errored      bool
+	}{
+		{"valid duration", `{"duration":"1ns"}`, false, false},
+		{"huge duration", `{"duration":"1000h"}`, false, true},
+		{"invalid duration", `{"duration":30}`, true, true},
```

Comment:
Instead of `duration`, let's make this `seconds` and just take an integer. concatenating a string in solidity to say essentially `"1000" + "h"` is really expensive, but generating a [number of seconds](https://solidity.readthedocs.io/en/v0.4.21/units-and-global-variables.html#time-units) can be represented like `1000 hours`.

Reply:
Alternatively, `{"duration":30}` could be valid, and just implied as seconds. I like the flexibility, but it may surprise some people.